src/main/java/org/opensearch/knn/index/engine/validation/ValidationUtil.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.engine.validation;
+
+import org.opensearch.common.ValidationException;
+
+public final class ValidationUtil {
```

Comment:
we can use @UtilClass , which make this class by default final. and we can get rid of Util suffix

Reply:
I did see that the current status of UtilClass getting out of experimental was negative (https://projectlombok.org/features/experimental/UtilityClass). That being said, might hold off.